src/Microsoft.TestPlatform.ObjectModel/TestCase.cs

```diff
@@ -261,7 +262,13 @@ protected override void ProtectedSetPropertyValue(TestProperty property, object?
                 return;
 
             case "TestCase.Id":
-                Id = value is Guid guid ? guid : Guid.Parse((value as string)!);
+                Id = value is Guid guid
+                    ? guid
+#if NET7_0_OR_GREATER
+                    : Guid.Parse((value as string)!, CultureInfo.InvariantCulture);
```

Comment:
Ouch there is something fishy here. We should not be doing `as` cast to then bang, it's either nullable or we should use direct cast instead.